libs/dh/admin/data-access-api/src/lib/dh-admin-assignable-user-roles.store.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * @license
+ * Copyright 2020 Energinet DataHub A/S
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License2");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Injectable } from '@angular/core';
+import { Observable, Subject, switchMap, tap } from 'rxjs';
+import { ComponentStore, tapResponse } from '@ngrx/component-store';
+
+import { ErrorState, LoadingState } from '@energinet-datahub/dh/shared/data-access-api';
+import { MarketParticipantUserRoleHttp, UserRoleDto } from '@energinet-datahub/dh/shared/domain';
+
+interface State {
+  readonly requestState: LoadingState | ErrorState;
+  readonly assignableUserRoles: UserRoleDto[];
+}
+
+const initialState: State = {
+  requestState: LoadingState.INIT,
+  assignableUserRoles: [],
+};
+
+@Injectable({ providedIn: 'root' })
```

Comment:
question: is this going to be used in other places than `DhAssignableUserRolesComponent`? If not, the `providedIn: 'root'` part can be removed. Providing a service in `root` means that the service will be long-lived once instantiated, even if it's not needed anymore.

Alternatively, it can be provided in the component where it's needed and Angular will destroy it automatically when the component is destroyed.

It seems that it's used both in `DhInviteUserModalComponent` (indirectly) and `DhAssignableUserRolesComponent` (directly). Since `DhAssignableUserRolesComponent` is used inside `DhInviteUserModalComponent`, the store can be provided in `DhInviteUserModalComponent` and both components will have access to it. 

So it can be injected in `DhInviteUserModalComponent` without the need to find a component and call one of its methods.

Reply:
Copy / paste error :-)